mlx_graphs/data/batch.py

```diff
@@ -8,35 +7,38 @@
 
 class Batch(GraphData):
     """
-        Represents a batch data object describing a batch of graphs as one big (disconnected)
-        graph.
+    Represents a batch data object describing a batch of graphs as one big (disconnected)
+    graph.
 
     """
 
-    def __init__(self, node_features, edge_index, num_nodes) -> None:
+    def __init__(self, node_features, edge_index, cumsum) -> None:
```

Comment:
Missing batch_size arg

Reply:
What would batch_size do in this case ? isn't it directly inferred from the length of the graph list provided ? 

I'd have thinked that when creating a Dataset object there'd be a constructor that when a list of graphs is provided he would create n_batches, where n_batches = len(graphs) % batch_size ( so if we have a list of 10 graphs and specificy a batch_size of 3 we'd have 4 batches, 3 of them have each 3 graphs and the last only one )

Reply:
Yes you're right, sorry for the oversight, batch size will be located in a dataset or loader class!